### DISCUSSION TITLE
TypeScript Config for builtin Next-ESLint

### DISCUSSION THREAD

**Discussion by bitsmyth** [vercel/next.js#26652] [Help]
Hi, I have read the documentation and find it pretty cool that we have an own eslint extension for nextjs. Therefore I find it kind of disapointing that there is no implementation for typescript users. Can someone provide an working example please?

**Comment by housseindjirdeh** [comment]
`@typescript-eslint/parser` is already supported and [used](https://github.com/vercel/next.js/blob/canary/packages/eslint-config-next/index.js#L37-L49) in all TS pages. So both TypeScript and ESLint are supported in Next.js :)

If you're referring to including additional type-specific rules - you can extend from the Next.js' built-in configuration as well as install and include [`@typescript-eslint/eslint-plugin`](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin):

```
{
  "extends": ["next", "plugin:@typescript-eslint/recommended"]
}
```

If you think this should be part of the documentation, let me know! But just like we don't enforce `eslint:recommended`, we've also left the choice up to the developer to include any TS-specific rules however they like :)
